std/typecons.d

```diff
@@ -804,11 +804,15 @@ if (distinctFieldNames!(Specs))
          *   $(LI A positive integer if the expression `v1 > v2` is true.)
          *   $(LI 0 if the expression `v1 == v2` is true.))
          */
-        int opCmp(R)(R rhs)
+        float opCmp(R)(R rhs)
         if (areCompatibleTuples!(typeof(this), R, "<"))
         {
             static foreach (i; 0 .. Types.length)
             {
+                if (field[i] != field[i] || rhs.field[i] != rhs.field[i])
```

Comment:
This is quite a pessimization, especially for user defined types. I would suggest:

1. Apply this check only for floating point types
2. Move it into the following `if`-block as `field[i] != rhs.field[i]` should be false if either operand is `NaN` (unless my FPN math is wrong)